src/devices/Mlx90640/Mlx90640.cs

```diff
@@ -0,0 +1,1072 @@
+            vdd = GetVdd(frameData);
+            ta = GetTa(frameData);
+
+            ta4 = (float)(ta + 273.15);
+            ta4 = ta4 * ta4;
+            ta4 = ta4 * ta4;
+            tr4 = (float)(tr + 273.15);
+            tr4 = tr4 * tr4;
+            tr4 = tr4 * tr4;
+            taTr = tr4 - (tr4 - ta4) / emissivity;
+
+            ktaScale = (float)(Math.Pow(2, _ktaScale));
+            kvScale = (float)(Math.Pow(2, _kvScale));
+            alphaScale = (float)(Math.Pow(2, _alphaScale));
+
+            alphaCorrR[0] = (float)(1 / (1 + _ksTo[0] * 40));
+            alphaCorrR[1] = 1;
+            alphaCorrR[2] = (float)((1 + _ksTo[1] * _ct[2]));
+            alphaCorrR[3] = (float)(alphaCorrR[2] * (1 + _ksTo[2] * (_ct[3] - _ct[2])));
+
+            //------------------------- Gain calculation -----------------------------------
+            gain = frameData[778];
+            if (gain > 32767)
+            {
+                gain = gain - 65536;
+            }
+
+            gain = (float)(_gainEE / gain);
+
+            //------------------------- To calculation -------------------------------------
+            mode = (byte)((frameData[832] & 0x1000) >> 5);
+
+            irDataCP[0] = frameData[776];
+            irDataCP[1] = frameData[808];
+            for (int i = 0; i < 2; i++)
+            {
+                if (irDataCP[i] > 32767)
+                {
+                    irDataCP[i] = irDataCP[i] - 65536;
+                }
+
+                irDataCP[i] = irDataCP[i] * gain;
+            }
+
+            irDataCP[0] = (float)(irDataCP[0] - _cpOffset[0] * (1 + _cpKta * (ta - 25)) * (1 + _cpKv * (vdd - 3.3)));
+            if (mode == _calibrationModeEE)
+            {
+                irDataCP[1] = (float)(irDataCP[1] - _cpOffset[1] * (1 + _cpKta * (ta - 25)) * (1 + _cpKv * (vdd - 3.3)));
+            }
+            else
+            {
+                irDataCP[1] = (float)(irDataCP[1] - (_cpOffset[1] + _ilChessC[0]) * (1 + _cpKta * (ta - 25)) * (1 + _cpKv * (vdd - 3.3)));
+            }
+
+            for (int pixelNumber = 0; pixelNumber < 768; pixelNumber++)
+            {
+                ilPattern = (sbyte)(pixelNumber / 32 - (pixelNumber / 64) * 2);
+                chessPattern = (byte)(ilPattern ^ (pixelNumber - (pixelNumber / 2) * 2));
+                conversionPattern = (byte)(((pixelNumber + 2) / 4 - (pixelNumber + 3) / 4 + (pixelNumber + 1) / 4 - pixelNumber / 4) * (1 - 2 * ilPattern));
+
+                if (mode == 0)
+                {
+                    pattern = (byte)ilPattern;
+                }
+                else
+                {
+                    pattern = chessPattern;
+                }
+
+                if (pattern == frameData[833])
+                {
+                    irData = frameData[pixelNumber];
+                    if (irData > 32767)
+                    {
+                        irData = irData - 65536;
+                    }
+
+                    irData = irData * gain;
```

Comment:
```suggestion
                    irData = gain * (short)frameData[pixelNumber];
```

also I suggest to use local variables closer to where you use them and do not reuse them later